migrations/0002_create-profile.sql

```diff
@@ -0,0 +1,67 @@
+-- Migration number: 0002 	 2025-01-30T00:58:21.685Z
+
+DROP TABLE IF EXISTS profile;
+
+-- A person's profile inside the database
+-- It should contain no sensitive information
+CREATE TABLE IF NOT EXISTS profile (
+	-- The UUID, stored as text
+	id TEXT NOT NULL UNIQUE COLLATE BINARY,
+	-- The person's email,
+	-- It is used as a reference for the user
+	-- And should be unique inside the system
+	email TEXT NOT NULL UNIQUE,
+	-- Schema version to use
+	schemaVersion INTEGER NOT NULL DEFAULT 1,
+	-- The person's name, or how they like to be identified
+	name TEXT NOT NULL,
+	-- A text blurb the person can provide about themselves
+	description TEXT,
+	-- A flag indicating if the user is based on the Grater Toronto Area (GTA)
+	isBasedOnGTA INTEGER NOT NULL DEFAULT 1 CHECK(isBasedOnGTA IN (0, 1)),
+	-- A flag indicating if the user is available to join local/in-person events
+	canJoinLocalEvents INTEGER NOT NULL DEFAULT 1 CHECK(canJoinLocalEvents IN (0, 1)),
+	-- The pronouns the person identifies with
+	pronouns TEXT,
+	-- The person's birthday, saved as month and day only
+	-- In the format: YYYY-MM
+	birthday TEXT,
+	-- The user avatar reference,
+	avatar TEXT,
+	-- The avatar may be a url, or an id for a file hosted on the platform
+	avatarSource TEXT DEFAULT 'id' CHECK(avatarSource IS NULL OR avatarSource IN ('id', 'url')),
+	-- The date this person has joined Toronto JS
+	happenedAt TEXT NOT NULL,
```

Comment:
As someone FE focused more like a question for my understanding. 

We store dates as text and not as a particular date format on BE? Similar question where ever you are mentioning date but type is TEXT

Reply:
This is a limitation to the [data types supported by SQLite](https://sqlite.org/datatype3.html#affinity_name_examples), it does understand if you set a column as `DATE`, but it will be converted to something else.

I'm thinking here, maybe we should use the specific data types...

Reply:
I was reviewing the code and checking for SQLite data types and remembered _why_ it is a `TEXT` instead of `DATE`: it is supposed to be an ISO timestamp, not simply a `DATETIME` like in standard SQL.

I ran some tests and apparently ISO timestamps work out of the box with the `DATETIME` type in SQLite, so I've updated the types and added more context to the comments.